<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>930</width>
    <height>598</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>40</y>
     <width>77</width>
     <height>135</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="basicset1">
    <item row="0" column="0">
     <widget class="QComboBox" name="portComboBox"/>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="openPortButton">
      <property name="text">
       <string>打开串口</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QComboBox" name="baudComboBox">
      <item>
       <property name="text">
        <string>2400</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>4800</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>9600</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>19200</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>57600</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>115200</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QPushButton" name="clearTextButton">
      <property name="text">
       <string>清空数据</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QPushButton" name="clearSendButton">
      <property name="text">
       <string>清除发送框</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="sendButton_9">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>350</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>传感器选择</string>
   </property>
   <property name="iconSize">
    <size>
     <width>100</width>
     <height>16</height>
    </size>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>190</y>
     <width>91</width>
     <height>132</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="basicset2">
    <item>
     <widget class="QLabel" name="label_9">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="text">
       <string>设备地址</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="recvTextEdit_8">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="sendButton_8">
      <property name="text">
       <string>获取当前地址</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="readdata_Button">
      <property name="text">
       <string>读取数据</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="sendButton_7">
      <property name="text">
       <string>停止读取</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>10</y>
     <width>156</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>设备校准（非专业人士勿动）</string>
   </property>
  </widget>
  <widget class="QWidget" name="formLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>260</y>
     <width>171</width>
     <height>84</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout">
    <item row="0" column="0">
     <widget class="QCheckBox" name="checkBox_1">
      <property name="text">
       <string>土壤温度</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QCheckBox" name="checkBox_4">
      <property name="text">
       <string>土壤氮</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QCheckBox" name="checkBox_5">
      <property name="text">
       <string>土壤磷</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QCheckBox" name="checkBox_3">
      <property name="text">
       <string>土壤电导率</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QCheckBox" name="checkBox_6">
      <property name="text">
       <string>土壤钾</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QCheckBox" name="checkBox_2">
      <property name="text">
       <string>土壤湿度</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QCheckBox" name="checkBox_7">
      <property name="text">
       <string>土壤PH</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="formLayoutWidget">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>40</y>
     <width>261</width>
     <height>542</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="17" column="2">
     <widget class="QPushButton" name="sendButton_16">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QPushButton" name="sendButton_12">
      <property name="text">
       <string>电导率零点</string>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">buttonGroup</string>
      </attribute>
     </widget>
    </item>
    <item row="16" column="2">
     <widget class="QPushButton" name="sendButton_15">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="9" column="1">
     <widget class="QPushButton" name="sendButton_4">
      <property name="text">
       <string>电导率8000</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="sendButton_2">
      <property name="text">
       <string>湿度零点</string>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">buttonGroup</string>
      </attribute>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QTextEdit" name="recvTextEdit_24">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_25">
      <property name="text">
       <string>湿度80% AD</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QTextEdit" name="recvTextEdit_23">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="14" column="0">
     <widget class="QLabel" name="label_17">
      <property name="text">
       <string>电导率8000 AD</string>
      </property>
     </widget>
    </item>
    <item row="17" column="1">
     <widget class="QTextEdit" name="recvTextEdit_12">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QTextEdit" name="recvTextEdit_21">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="17" column="0">
     <widget class="QLabel" name="label_19">
      <property name="text">
       <string>PH中性 AD</string>
      </property>
     </widget>
    </item>
    <item row="14" column="1">
     <widget class="QTextEdit" name="recvTextEdit_9">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="12" column="0">
     <widget class="QLabel" name="label_14">
      <property name="text">
       <string>电导率1500 AD</string>
      </property>
     </widget>
    </item>
    <item row="15" column="2">
     <widget class="QPushButton" name="sendButton_10">
      <property name="text">
       <string>PH酸性4.00</string>
      </property>
     </widget>
    </item>
    <item row="13" column="1">
     <widget class="QTextEdit" name="recvTextEdit_17">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="16" column="1">
     <widget class="QTextEdit" name="recvTextEdit_10">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="19" column="0" colspan="3">
     <widget class="QPushButton" name="sendButton_18">
      <property name="text">
       <string>一键获取当前设置值</string>
      </property>
     </widget>
    </item>
    <item row="12" column="1">
     <widget class="QTextEdit" name="recvTextEdit_14">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="8" column="2">
     <widget class="QPushButton" name="sendButton_14">
      <property name="text">
       <string>电导率1500</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QTextEdit" name="recvTextEdit_22">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_22">
      <property name="text">
       <string>湿度零点 AD</string>
      </property>
     </widget>
    </item>
    <item row="18" column="2">
     <widget class="QPushButton" name="sendButton_17">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="18" column="1">
     <widget class="QTextEdit" name="recvTextEdit_11">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QPushButton" name="sendButton_5">
      <property name="text">
       <string>电导率500</string>
      </property>
     </widget>
    </item>
    <item row="13" column="0">
     <widget class="QLabel" name="label_15">
      <property name="text">
       <string>电导率3000 AD</string>
      </property>
     </widget>
    </item>
    <item row="15" column="0">
     <widget class="QPushButton" name="sendButton_6">
      <property name="text">
       <string>PH碱性9.18</string>
      </property>
     </widget>
    </item>
    <item row="15" column="1">
     <widget class="QPushButton" name="sendButton_11">
      <property name="text">
       <string>PH中性6.86</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="sendButton_21">
      <property name="text">
       <string>湿度20%</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_24">
      <property name="text">
       <string>湿度50% AD</string>
      </property>
     </widget>
    </item>
    <item row="11" column="2">
     <widget class="QPushButton" name="pushButton">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="13" column="2">
     <widget class="QPushButton" name="pushButton_3">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_23">
      <property name="text">
       <string>湿度20% AD</string>
      </property>
     </widget>
    </item>
    <item row="11" column="1">
     <widget class="QTextEdit" name="recvTextEdit_13">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="18" column="0">
     <widget class="QLabel" name="label_20">
      <property name="text">
       <string>PH酸性 AD</string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QPushButton" name="sendButton_13">
      <property name="text">
       <string>电导率3000</string>
      </property>
     </widget>
    </item>
    <item row="12" column="2">
     <widget class="QPushButton" name="pushButton_2">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="16" column="0">
     <widget class="QLabel" name="label_18">
      <property name="text">
       <string>PH碱性 AD</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="sendButton_23">
      <property name="text">
       <string>湿度80%</string>
      </property>
     </widget>
    </item>
    <item row="11" column="0">
     <widget class="QLabel" name="label_10">
      <property name="text">
       <string>电导率500 AD</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="sendButton_22">
      <property name="text">
       <string>湿度50%</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="sendButton_3">
      <property name="text">
       <string>湿度顶点</string>
      </property>
      <attribute name="buttonGroup">
       <string notr="true">buttonGroup</string>
      </attribute>
     </widget>
    </item>
    <item row="14" column="2">
     <widget class="QPushButton" name="pushButton_4">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="label_26">
      <property name="text">
       <string>湿度顶点 AD</string>
      </property>
     </widget>
    </item>
    <item row="7" column="1">
     <widget class="QTextEdit" name="recvTextEdit_25">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>40</y>
     <width>171</width>
     <height>213</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="dataread">
    <item row="5" column="1">
     <widget class="QTextEdit" name="recvTextEdit_5">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QTextEdit" name="recvTextEdit_2">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QTextEdit" name="recvTextEdit">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>土壤磷</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>土壤电导率</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QTextEdit" name="recvTextEdit_4">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QTextEdit" name="recvTextEdit_6">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>土壤湿度</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>土壤氮</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QTextEdit" name="recvTextEdit_3">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>土壤钾</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="label_8">
      <property name="text">
       <string>土壤PH</string>
      </property>
     </widget>
    </item>
    <item row="7" column="1">
     <widget class="QTextEdit" name="recvTextEdit_7">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>土壤温度</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>40</y>
     <width>171</width>
     <height>138</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="dataread_2">
    <item row="1" column="1">
     <widget class="QTextEdit" name="recvTextEdit_16">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QTextEdit" name="recvTextEdit_19">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_12">
      <property name="text">
       <string>土壤湿度ADC</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_11">
      <property name="text">
       <string>土壤电导率ADC</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QTextEdit" name="recvTextEdit_15">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_13">
      <property name="text">
       <string>土壤PH ADC</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QTextEdit" name="recvTextEdit_20">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_16">
      <property name="text">
       <string>土壤温度ADC</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>190</y>
     <width>171</width>
     <height>71</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="dataread_3">
    <item row="0" column="1">
     <widget class="QTextEdit" name="recvTextEdit_18">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>25</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="label_21">
      <property name="text">
       <string>地址</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="sendButton_19">
      <property name="text">
       <string>SET</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="3">
     <widget class="QPushButton" name="sendButton_20">
      <property name="text">
       <string>恢复为默认地址</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
